llarp/router/router.cpp

```diff
@@ -227,14 +229,97 @@ namespace llarp
       LogError(rcfile, " contains invalid RC");
   }
 
+  static size_t
+  RecvIdentKey(char *ptr, size_t, size_t nmemb, void *userdata)
+  {
+    for(size_t idx = 0; idx < nmemb; idx++)
+      static_cast< std::vector< char > * >(userdata)->push_back(ptr[idx]);
+    return nmemb;
+  }
+
   bool
   Router::EnsureIdentity()
   {
     if(!EnsureEncryptionKey())
       return false;
-    if(usingSNSeed)
-      return llarp_loadServiceNodeIdentityKey(ident_keyfile, _identity);
-
+    if(whitelistRouters)
+    {
+#if defined(ANDROID) || defined(IOS)
+      LogError("running a service node on mobile device is not possible.");
+      return false;
+#else
+      CURL *curl = curl_easy_init();
+      if(curl)
+      {
+        CURLcode res;
+        std::stringstream ss;
+        ss << "http://" << lokidRPCAddr << "/json_rpc";
+        const auto url = ss.str();
```

Comment:
Simpler here just to use string concatenation: `const auto url = "http://" + lokidRPCAddr + "/json_rpc";`

Reply:
(Assuming lokidRPCAddr is already a `std::string`; otherwise wrap one of these in a `std::string{...}` to make it work)